neqo-bin/src/client/mod.rs

```diff
@@ -376,7 +376,7 @@ trait Client {
     fn process_output(&mut self, now: Instant) -> Output;
     fn process_multiple_input<'a, I>(&mut self, dgrams: I, now: Instant)
     where
-        I: IntoIterator<Item = &'a Datagram>;
+        I: IntoIterator<Item = Datagram<&'a [u8]>>;
```

Comment:
Is it possible to make this `IntoIterator<Item = Datagram<impl AsRef<&'a [u8]>>>` instead?  (Or `Borrow`, as below.)   Then you could maybe drop the borrow calls in a few places. (I get that the multi interfaces are probably not that well suited to calling with `Vec<u8>`, but might as well be consistent throughout.

Reply:
> Is it possible to make this `IntoIterator<Item = Datagram<impl AsRef<&'a [u8]>>>` instead?

Interesting idea. It is possible. I sketched it in https://github.com/mxinden/neqo/commit/febd6f3082303530a562d3cfdddf1a0f48be6da1. 

That said, is this additional indirection worth the complexity it adds to the signature?

In my eyes, given that it is not strictly needed (e.g. no performance or functionality gain), and given that there are few `.borrow()` calls only, I don't think it is worth it.

> (Or `Borrow`, as below.)

Unfortunately, one can not use `Borrow`, given the returned reference to a temporary object. See comment below (https://github.com/mozilla/neqo/pull/2187#discussion_r1806044720).

Reply:
Recent commits implement the above sketch in full. See also https://github.com/mozilla/neqo/pull/2187#issuecomment-2426040259.